feat: FIT-750: Update Agreement Selected UI

yyassi-heartex opened the pull request:
This pull request improves the handling of the Agreement filter dropdown in the data manager table. The main focus is to ensure that the dropdown closes automatically after saving changes and to provide a way for the header cell to trigger closing the dropdown.

**Dropdown behavior improvements:**

* Added an `onClose` callback to `AgreementSelected.HeaderCell`, allowing the header cell to close the dropdown when needed. [[1]](diffhunk://#diff-0c4e72d7ef0760ff179a3d2c412f66de6ab631378cd7e1220050332292c3cfd5L37-R42) [[2]](diffhunk://#diff-51ab382cac0e258f4e4a21289401f0fda6d405059dae8badcc56b0c9e2bc8d5dL87-R91)
* Updated the `onSave` handler in `TableHead.jsx` to automatically close the dropdown after saving agreement filters.

yyassi-heartex committed:
feat: FIT-750: Update Agreement Selected UI

yyassi-heartex committed:
linting

yyassi-heartex committed:
updating to the new agreement_selected structure

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
linting

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
resolving an issue with reopening the popover after closing requiring 2 clicks

yyassi-heartex committed:
updating default value

yyassi-heartex committed:
we now have a min width on agreement selected

yyassi-heartex committed:
lint

Commit:
update migration

Commit:
use service queue for migration

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
we now fire an event with the invoke ok action is completed

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

mcanu commented on label_studio/data_manager/migrations/0017_update_agreement_selected_to_nested_structure.py:
```diff
@@ -0,0 +1,103 @@
+
+        new_data = deepcopy(data)
+        # Always use the new nested structure
+        new_data['agreement_selected'] = {
+            'annotators': {'all': True, 'ids': []},
+            'models': {'all': True, 'ids': []},
+            'ground_truth': False
+        }
```
We don't care for existing configurations right? This isn't fully released yet, so I think its okay

yyassi-heartex replied:
it was enabled for everyone for a few days - we had to turn it off after a few days - so the migration is needed

bmartel replied:
It was left on for some people, so it definitely will need to migrate from old to new.

mcanu replied:
@hakan458 can we check how many people have it configured? I feel that its okay to reset it them for them, but not certain.

hakan458 replied:
There are 2 customers with it enabled that we have communicated to already 👍🏼

AndrejOros commented:
/fm sync 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18463145003)

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18463145003

AndrejOros committed:
Merge branch 'develop' into 'fb-fit-750/updated-ui'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18463145003

yyassi-heartex committed:
Merge branch 'develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
fixed tooltip button visuals

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

yyassi-heartex committed:
lint

AndrejOros commented:
/git merge 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18560130880)
> Successfully merged:  3 files changed, 31 insertions(+), 4 deletions(-)

AndrejOros committed:
Merge branch 'develop' into 'fb-fit-750/updated-ui'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18560130880

yyassi-heartex committed:
AgreementSelected columns should behave as a number

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui

AndrejOros commented:
/git merge 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18584508869)
> Successfully merged:  create mode 100644 web/libs/ui/src/lib/modal/index.ts

AndrejOros committed:
Merge branch 'develop' into 'fb-fit-750/updated-ui'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18584508869

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-750/updated-ui
